build.sbt

```diff
@@ -357,7 +377,25 @@ lazy val connectClient = (project in file("spark-connect/client"))
       } else {
         dest.get()
       }
-    }.taskValue
+    }.taskValue,
+    Compile / packageBin := assembly.value,
+    (assembly / test) := { },
+    (assembly / logLevel) := Level.Info,
+    // Exclude `scala-library` from assembly.
+    (assembly / assemblyPackageScala / assembleArtifact) := false,
+    (assembly / assemblyShadeRules) := Seq(
+      ShadeRule.rename("com.google.**" -> "org.deltaproject.connect.com.google.@1").inAll,
+      ShadeRule.rename("io.grpc.**" -> "org.deltaproject.connect.io.grpc.@1").inAll,
```

Comment:
By inspecting the jar files of the 3 Delta Connect projects after running `publishM2`, I believe the shading rule are not propagated to dependent projects, so even if only `connectCommon` uses `io.grpc`, we still need to add shading rule `io.grpc` to `connectClient` since `connectClient` depends on `connectCommon` so `connectClient` includes `connectCommon` codes and we need to shade those codes too.